### DISCUSSION TITLE
About CRUD Permissions

### DISCUSSION THREAD

**Discussion by Faridalim** [awtkns/fastapi-crudrouter#47] [Q&A] [answered]
Hi, could you guide me how to implement fastapi-crudrouter with JWT scope? Or maybe a way to limit some CRUD routes to some roles? Thank you very much

**Comment by awtkns** [comment, marked as answer]
Hi there. Currently you can limit CRUD routes to some roles by adding a dependency ([Docs](https://fastapi-crudrouter.awtkns.com/dependencies/)) to your router. eg:

```python
from fastapi import FastAPI, testclient, Depends
from fastapi.security import OAuth2PasswordBearer
from fastapi_crudrouter import MemoryCRUDRouter

app = FastAPI()
oauth2_scheme = OAuth2PasswordBearer(tokenUrl="auth/token")

def token_auth(token: str):
    if not token:
        raise HTTPException(401, "Invalid token")

router = MemoryCRUDRouter(schema=Potato, dependencies=[Depends(token_auth)])
app.include_router(router)
``` 

If you are trying to limit just individual routes to certain scopes (eg: just the delete all route), there is currently no way to do that,  However it will be available in a future release.

#37

**Reply by Faridalim** [reply]
Thank you for the reply.. Wow, the implementation is so easy, thank you.. Okay, i will be waiting for the update :)

**Reply by awtkns** [reply]
The discussion on the feature can be found in #37

**Reply by awtkns** [reply]
Hi @Faridalim, this feature (dependencies for individual routes) has now been released in the latest version v0.7.

**Reply by Faridalim** [reply]
Super.. thank you @awtkns, I will check it
